src/player/Player.php

```diff
@@ -1685,15 +1685,24 @@ public function pickBlock(Vector3 $pos, bool $addTileNBT) : bool{
 			return true;
 		}
 
-		$item = $block->getPickedItem($addTileNBT);
+		$pickedItem = $block->getPickedItem($addTileNBT);
 
-		$ev = new PlayerBlockPickEvent($this, $block, $item);
-		$existingSlot = $this->inventory->first($item);
-		if($existingSlot === -1 && $this->hasFiniteResources()){
-			$ev->cancel();
-		}
+		$ev = new PlayerBlockPickEvent($this, $block, $pickedItem);
+		$checkExistingSlot = function(Item $item) use ($ev) : int{
```

Comment:
I don't think that's the best name. Maybe something like ``findExistingSlot`` would be more appropriate